stream-chat-android-offline/src/main/java/io/getstream/chat/android/offline/experimental/channel/logic/ChannelLogic.kt

```diff
@@ -148,6 +150,28 @@ internal class ChannelLogic(
             }
     }
 
+    override suspend fun onHideChannelRequest(channelType: String, channelId: String, clearHistory: Boolean) = setHidden(true)
+
+    override suspend fun onHideChannelResult(
+        result: Result<Unit>,
+        channelType: String,
+        channelId: String,
+        clearHistory: Boolean,
+    ) {
+        if (result.isSuccess) {
+            val cid = Pair(channelType, channelId).toCid()
```

Comment:
I think we should try to validate it earlier. It doesn't make sense to do an API call if cid is incorrect and then throw an exception in side effect function

Reply:
Do we need this check here? We are already doing that in precondition

Reply:
@amitkma ☝️

Reply:
Which check?

Reply:
my bad, we can resolve